roles/openshift_autoheal/files/template.yml

```diff
@@ -0,0 +1,198 @@
+---
+
+apiVersion: v1
+kind: Template
+metadata:
+  name: autoheal-template
+  annotations:
+    description: "Auto-heal service"
+    tags: "autoheal"
+
+parameters:
+- name: IMAGE
+  description: The name of the image.
+- name: CONFIG
+  description: The BASE64 encoded content of the configuration file.
+- name: SECRET
+  description: The BASE64 encoded secret used to encrypt OAuth session cookies.
+
+objects:
+
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    name: autoheal
+    labels:
+      app: autoheal
+
+- apiVersion: authorization.openshift.io/v1
+  kind: Role
+  metadata:
+    name: autoheal
+    labels:
+      app: autoheal
+  rules:
+  - apiGroups:
+    - ""
+    resources:
+    - secrets
+    resourceNames:
+    - autoheal-config
+    verbs:
+    - get
+
+- apiVersion: authorization.openshift.io/v1
+  kind: ClusterRole
```

Comment:
Are the SA/ClusterRole/ClusterRoleBindings the specific resources you were concerned would need split off into another file? I guess since those are non-namespaced resources we can cheat in this case 😄

Reply:
The objects that I was concerned about aren't needed now. They will be needed when we add custom API and etcd servers to persistently manage the healing rules (and other kinds of objects). Note that the problem will then be to have objects in other namespaces, in particular in the `kube-system` namespace. But as I said that is concert for the future, not now. When the time comes we will solve that, I guess, using two or more templates.